File: new_backend/src/models/subChapter.ts
import { Document, Schema, model } from "mongoose";

export interface ISubChapter extends Document {
    name: String;
    contents: [Schema.Types.ObjectId];
    chapterId: Schema.Types.ObjectId;
    adminApproval: Boolean,
    order:Number
}

const SubChapterSchema: Schema<ISubChapter> = new Schema({
    chapterId: {
        type: Schema.Types.ObjectId,
        ref: 'Chapter',
        required: [true, "Chapter Id is required"]
    },
    name: {
        type: String,
        required: [true, "Sub-chapter name is required"]
    },
    contents: {
        type: [{
            type: Schema.Types.ObjectId,
            ref: "SubChapterContent"
        }],
        default: []
    },
    adminApproval: {
        type: Boolean,
        default: true
    },
    order: {
        type: Number,
    }
},
    {
        timestamps: {
            createdAt: 'createdAt',
            updatedAt: 'updatedAt'
        }
    }
);

const SubChapter = model<ISubChapter>("SubChapter", SubChapterSchema);

export default SubChapter;

File: new_mobile/lib/features/chapter/domain/entities/sub_chapter.dart
import 'package:equatable/equatable.dart';

import 'entities.dart';

class SubChapter extends Equatable {
  final String id;
  final String chapterId;
  final String name;
  final List<Content> contents;
  final int? order;

  const SubChapter({
    required this.id,
    required this.chapterId,
    required this.name,
    required this.contents,
    this.order,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        contents,
        chapterId,
        order
      ];
}
